dd-sdk-android/src/test/kotlin/com/datadog/android/utils/forge/ForgeExt.kt

```diff
@@ -34,5 +42,47 @@ internal fun Forge.exhaustiveAttributes(): Map<String, Any?> {
         .toMap().toMutableMap()
     map[""] = anHexadecimalString()
     map[aWhitespaceString()] = anHexadecimalString()
-    return map
+
+    val filtered = map.filterKeys { it !in excludedKeys }
+
+    assumeDifferenceIsNoMore(filtered.size, map.size, filterThreshold)
+
+    return filtered
+}
+
+/**
+ * Creates a map just like [Forge#aMap], but it won't include given keys.
+ * @param excludedKeys Keys to exclude from generated map.
+ * @param filterThreshold Max ratio of keys removed from originally generated map. If ratio
+ * is more than that, [Assume] mechanism will be used.
+ */
+internal fun <K, V> Forge.aFilteredMap(
+    size: Int = -1,
+    excludedKeys: Set<K>,
+    filterThreshold: Float = 0.5f,
+    forging: Forge.() -> Pair<K, V>
+): Map<K, V> {
+
+    val base = aMap(size, forging)
+
+    val filtered = base.filterKeys { it !in excludedKeys }
+
+    if (base.isNotEmpty()) {
+        assumeDifferenceIsNoMore(filtered.size, base.size, filterThreshold)
+    }
+
+    return filtered
+}
+
+private fun assumeDifferenceIsNoMore(result: Int, base: Int, maxDifference: Float) {
+
+    check(result <= base) {
```

Comment:
is that even possible ?

Reply:
well, maybe for very small maps, but even in that case chance is not that big. anyway, I decided to leave it like that, so that if we fail to produce the necessary test prerequisites we had some traces of what happened at least.